Bottom: 202cc4ccec12521215fc5c6923d087591b0618f0
Top:    a20b33583c60c26886459542d9133ce850408676
Author: Jon Smirl <jonsmirl@gmail.com>
Date:   2009-04-26 13:56:25 -0400

Refresh of mpc5200-dma-split

---

diff --git a/sound/soc/fsl/mpc5200_dma.c b/sound/soc/fsl/mpc5200_dma.c
index f378b99..0034db3 100644
--- a/sound/soc/fsl/mpc5200_dma.c
+++ b/sound/soc/fsl/mpc5200_dma.c
@@ -343,9 +343,11 @@ static const struct snd_pcm_hardware psc_dma_pcm_hardware = {
 
 static int psc_dma_pcm_open(struct snd_pcm_substream *substream)
 {
+	struct snd_pcm_runtime *runtime = substream->runtime;
 	struct snd_soc_pcm_runtime *rtd = substream->private_data;
 	struct psc_dma *psc_dma = rtd->dai->cpu_dai->private_data;
 	struct psc_dma_stream *s;
+	int ret;
 
 	dev_dbg(psc_dma->dev, "psc_dma_pcm_open(substream=%p)\n", substream);
 
@@ -356,6 +358,13 @@ static int psc_dma_pcm_open(struct snd_pcm_substream *substream)
 
 	snd_soc_set_runtime_hwparams(substream, &psc_dma_pcm_hardware);
 
+	ret = snd_pcm_hw_constraint_integer(runtime,
+		SNDRV_PCM_HW_PARAM_PERIODS);
+	if (ret < 0) {
+		dev_err(substream->pcm->card->dev, "invalid buffer size\n");
+		return ret;
+	}
+
 	s->stream = substream;
 	return 0;
 }
